x-pack/plugins/infra/public/pages/metrics/page_template.tsx

```diff
@@ -20,5 +23,33 @@ export const MetricsPageTemplate: React.FC<LazyObservabilityPageTemplateProps> =
     },
   } = useKibanaContextForPlugin();
 
-  return <PageTemplate {...pageTemplateProps} />;
+  const tutorialLinkProps = useLinkProps({
+    app: 'home',
+    hash: '/tutorial_directory/metrics',
+  });
+
+  const integrationLinkProps = useLinkProps({
+    app: 'integrations/browse',
+  });
+
+  const { metricIndicesExist } = useContext(Source.Context);
+  const noDataConfig: KibanaPageTemplateProps['noDataConfig'] = metricIndicesExist
+    ? undefined
+    : {
+        solution: 'Observability',
+        pageTitle: 'Set up Metrics for Observability!',
```

Comment:
**Metrics**

Reply:
I think you can use:
```ts
const { metricIndicesExist } = useSource();
```
and just import `useSource` instead of `Source` from that same file.

But they both seem to be doing the same thing. @simianhacker @neptunian could you just confirm that this will correctly return whether metrics indices exist at this point in the app?

Reply:
Looks like `useSource` requires a parameter of `sourceId`. 
<img width="601" alt="Screen Shot 2021-08-12 at 11 11 46 AM" src="https://user-images.githubusercontent.com/549577/129221822-0ee2496e-46c7-4f28-8bd4-7a6973a40885.png">

Reply:
Well, that's what I want to confirm. 😆

Reply:
@weltenwort if you have a minute can you help me confirm whether this context access is expected to work without designating the source ID? I don't want to tell Caroline to "just use 'default' here" because that seems bad, but I wasn't able to figure out how we are supposed to interact with this...